component/src/main/java/io/siddhi/extension/store/elasticsearch/sink/ElasticsearchSink.java

```diff
@@ -0,0 +1,287 @@
+                                "in the elasticsearch server and saves the custom json document."
+                )
+        }
+)
+
+public class ElasticsearchSink extends Sink {
+
+    private static final Logger logger = Logger.getLogger(ElasticsearchSink.class);
+    private ElasticsearchConfigs elasticsearchConfigs;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{String.class};
+    }
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[0];
+    }
+
+    @Override
+    protected StateFactory init(StreamDefinition streamDefinition, OptionHolder optionHolder,
+                                ConfigReader sinkConfigReader, SiddhiAppContext siddhiAppContext) {
+        ElasticsearchConfigs.setLogger(logger);
+        elasticsearchConfigs = new ElasticsearchConfigs(this);
+        elasticsearchConfigs.init(streamDefinition, sinkConfigReader, siddhiAppContext);
+        return null;
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions dynamicOptions, State state)
+            throws ConnectionUnavailableException {
+        SiddhiIndexRequest siddhiIndexRequest = new SiddhiIndexRequest(payload, dynamicOptions,
+                elasticsearchConfigs.getIndexName());
+        try {
+            JsonElement parse = new JsonParser().parse((String) payload);
+            if (parse.isJsonArray()) {
+                JsonArray jsonArray = parse.getAsJsonArray();
+                for (JsonElement jsonElement : jsonArray) {
+                    siddhiIndexRequest.source(jsonElement.toString(), XContentType.JSON);
+                    elasticsearchConfigs.getBulkProcessor().add(siddhiIndexRequest);
+                }
+                logger.debug(payload + " has been successfully added.");
+            } else if (parse.isJsonObject()) {
+                siddhiIndexRequest.source(parse.getAsJsonObject().toString(), XContentType.JSON);
+                elasticsearchConfigs.getBulkProcessor().add(siddhiIndexRequest);
+            }
+        } catch (JsonSyntaxException e) {
+            throw new ElasticsearchEventSinkException("Invalid json document, Please recheck the json mapping at" +
+                    "the '@payload'.", e);
+        }
+    }
+
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+        if (elasticsearchConfigs.getIndexName() != null && !elasticsearchConfigs.getIndexName().isEmpty()) {
+            createIndex(elasticsearchConfigs.getDefinitionId());
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        try {
+            elasticsearchConfigs.getRestHighLevelClient().close();
+        } catch (IOException e) {
+            throw new ElasticsearchEventSinkException("Error while closing the connection.", e);
```

Comment:
Give more context to the error message.

Reply:
Fixed in https://github.com/siddhi-io/siddhi-store-elasticsearch/pull/41/commits/8887cc35f24420f2482087ac0c3b1e1aea2e6ccf commit